charms/istio-pilot/src/charm.py

```diff
@@ -1057,5 +1082,16 @@ def _xor(a, b):
         return False
 
 
+def _is_hostname(value: Optional[str]) -> bool:
+    """Return True if the given value is a valid hostname, False otherwise or if it's None."""
+    if value is None:
+        return False
+    try:
+        ipaddress.ip_address(value)
+        return False
+    except ValueError:
+        return bool(value)
```

Comment:
I don't know if this really matters, but this test is really `is_not_ipaddress` rather than `is_hostname`.  For example, these are not valid hostnames but pass:
* `_is_hostname("*")` returns `True`
* `_is_hostname("a"*1000)` returns `True`

Reply:
If its important, [this](https://stackoverflow.com/a/33214423) might be a more complete solution.  If it isn't important, renaming the test to `is_not_ipaddress` (or `is_ipaddress`) would be clearer

Reply:
I think I will rename it to `is_ipaddress`, it will be more clear that way as we only are checking that.